scripts/support/runner/jlink.py

```diff
@@ -124,15 +125,18 @@ def flash(self, **kwargs):
         lines.append('g') # Start the CPU
         lines.append('q') # Close the connection and quit
 
-        with tempfile.NamedTemporaryFile(suffix='.jlink') as f:
-            f.writelines(bytes(line + '\n', 'utf-8') for line in lines)
-            f.flush()
+        # Don't use NamedTemporaryFile: the resulting file can't be
+        # opened again on Windows.
+        with tempfile.TemporaryDirectory(suffix='jlink') as d:
+            fname = os.path.join(d, 'runner.jlink')
+            with open(fname, 'wb') as f:
+                f.writelines(bytes(line + '\n', 'utf-8') for line in lines)
 
             cmd = ([self.commander] +
                    ['-if', self.iface,
                     '-speed', self.speed,
                     '-device', self.device,
-                    '-CommanderScript', f.name])
+                    '-CommanderScript', shlex.quote(fname)])
```

Comment:
Note that shlex.quote is necessary since "There is [...] no guarantee that the generated filename will have any nice properties, such as not requiring quoting when passed to external commands via os.popen()."

https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp

(The same rules apply to mkdtemp, which is what TemporaryDirectory uses.)

This was always an issue, but might as well fix it now.